pkg/body_extractor/grpc_payload.go

```diff
@@ -117,30 +138,152 @@ func fieldFromProtoMessage(msg []byte, tagIndex int) (string, error) {
 	if err := dynamicMsgKey.Unmarshal(msg); err != nil {
 		return "", err
 	}
-	val, err := dynamicMsgKey.TryGetFieldByNumber(tagIndex)
+
+	val, err := RunQuery(dynamicMsgKey, parsedQuery)
 	if err != nil {
 		return "", err
 	}
 	return val.(string), nil
 }
 
-// should only be built once
-var genericProtoCache *desc.MessageDescriptor
+func buildPayloadProto(queries []Query) (*desc.MessageDescriptor, error) {
```

Comment:
building this at every call doesn't feel like the best idea, we should cache it.

Reply:
makes sense, doing it

Reply:
done